packages/medusa/src/utils/repository.ts

```diff
@@ -35,62 +40,93 @@ export async function queryEntityWithIds<T extends ObjectLiteral>({
     qb: SelectQueryBuilder<T>,
     alias: string,
     toplevel: string
-  ) => boolean)[]
+  ) => false | undefined)[]
 }): Promise<T[]> {
   const alias = repository.metadata.name.toLowerCase()
   return await Promise.all(
-    Object.entries(groupedRelations).map(async ([toplevel, rels]) => {
-      let querybuilder = repository.createQueryBuilder(alias)
+    Object.entries(groupedRelations).map(
+      async ([toplevel, topLevelRelations]) => {
+        let querybuilder = repository.createQueryBuilder(alias)
 
-      if (select && select.length) {
-        querybuilder.select(select.map((f) => `${alias}.${f as string}`))
-      }
+        if (select?.length) {
+          querybuilder.select(
+            (select as string[])
+              .filter(function (s) {
+                return s.startsWith(toplevel) || !s.includes(".")
+              })
+              .map((column) => {
+                // In case the column is the toplevel relation, we need to replace the dot with a double underscore
+                if (column.includes(toplevel)) {
+                  return topLevelRelations.some((rel) => column.includes(rel))
+                    ? column.replace(positiveLookaheadDotReplacer, "__")
```

Comment:
oh my god, is this a typeorm requirement? :o Might be worth checking if they have an exported util.

Reply:
It is to be aligned with the custom joining bellow where we replace with `__` to prevent auto aliasing. I don't touch that part to prevent the scope to grow ahah

Reply:
also, I ve tested to not replace the `.` by `__` but then typeorm becore confused and use the alias properly in some cases but not in other cases which leads to some issues as it think we are doing a cross db query because the dot alias is not used as a string.